### PR TITLE
[[SQUASH MERGE]] Teferi,Nahiri and Garruk dungeon + update other four dungeons

### PR DESCRIPTION
**Pull request by Simisays**
This PR went really outside of it's original scope :)

1. 3 New dungeons based on Planeswalkers: Teferi, Nahiri and Garruk
2. Putting the boss effects and reward in the command zone and generally changing the rewards a bit 
3. Cleaning up my older dungeons (map, enemies, decks ect, collision)
4. 4 new enemies with new sprites (Slime, Squirrel, fox and Kobold

Remade the slime boss dungeon and changed to a more fitting sprite. Also generally rebalanced some bosses/enemies by changing the decks and rewards.

### CODE REVIEW

**Comment by kevlahnota** [comment]
I like to update the custom cards something like this... 
image

**Comment by Simisays** [comment]
> I like to update the custom cards something like this (resized to 680p for lowend devices). image

Looks awesome, i'll get it replaced

**Comment by kevlahnota** [comment]
> Looks awesome, i'll get it replaced
no need to hurry though until the other PR is merged since your PR needs the start at the commandzone cards

File: forge-gui/res/adventure/Shandalar/custom_cards/slimefoots_slimy_staff.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Name:Slimefoot's Slimy Staff
+ManaCost:no cost
+Colors:black,green
+Types:Artifact Equipment
+A:AB$ MakeCard | Cost$ G PayShards<2> | Name$ Bayou | ActivationZone$ Command | Zone$ Hand | IsPresent$ Land.YouOwn | PresentZone$ Hand | PresentCompare$ EQ0 | GameActivationLimit$ 1 | SpellDescription$ Conjure a card named Bayou into your hand. Activate only if there are no land cards in your hand and only once each game. 
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Command | Execute$ TrigPutCounter | TriggerDescription$ At the beginning of your end step, put a spore counter on each fungus you control.
+SVar:TrigPutCounter:DB$ PutCounterAll | ValidCards$ Fungus.YouCtrl | CounterType$ SPORE | CounterNum$ 1
+Oracle:Provided by Slimefoot's Slimy Staff\nG,{M}{M}:Conjure a card named Bayou into your hand. Activate only if there are no land cards in your hand and only once each game. At the beginning of your end step, put a spore counter on each fungus you control.
```
Comment (kevlahnota) [inline forge-gui/res/adventure/Shandalar/custom_cards/slimefoots_slimy_staff.txt:8]: {M}{M}, {G}

File: forge-gui/res/adventure/Shandalar/custom_cards/kioras_bident.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+Name:Kiora's Bident
+ManaCost:no cost
+Types:Artifact Equipment
+S:Mode$ ReduceCost | EffectZone$ Command | ValidCard$ Crab.YouCtrl,Kraken.YouCtrl,Serpent.YouCtrl,Leviathan.YouCtrl,Octopus.YouCtrl | Type$ Spell | Activator$ You | Amount$ 1 | Description$ Kraken, Leviathan, Octopus, Crab and Serpent spells you cast cost {1} less.
+A:AB$ MakeCard | Cost$ U G PayShards<2> Exile<1/CARDNAME> | Conjure$ True | ActivationZone$ Command | AtRandom$ True | Spellbook$ Kiora, Behemoth Beckoner,Kiora, Master of the Depths,Kiora, the Crashing Wave,Kiora, the Tide's Fury
+Oracle:Provided by Kiora's Bident.\n Kraken, Leviathan, Octopus, Crab and Serpent spells you cast cost {1} less.\n {U}{G}, {M}{M}, Exile Kiora's Bident: Conjure a random Kiora planeswalker card into your hand.
```
Comment (kevlahnota) [inline forge-gui/res/adventure/Shandalar/custom_cards/kioras_bident.txt:6]: {M}{M}, {U}{G}

File: forge-gui/res/adventure/Shandalar/custom_cards/teferis_staff.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Name:Teferi's Staff
+ManaCost:no cost
+Types:Artifact Equipment
+A:AB$ Phases | Cost$ U W PayShards<2> | ValidTgts$ Permanent | ActivationZone$ Command | SpellDescription$ Target permanent phases out. (Treat it and anything attached to it as though they don't exist until its controllers's next turn.)
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Command | Execute$ TrigUntap | TriggerDescription$ At the beginning of your end step, untap target artifact or land you control.
+SVar:TrigUntap:DB$ Untap | ValidTgts$ Artifact.YouCtrl,Land.YouCtrl | TgtPrompt$ Select target artifact or land you control
+Oracle:
+Provided by Teferi's Staff.\n At the beginning of your end step, untap target artifact or land you control.\n{U{W}, {M}{M}, Exile Teferi's Staff: Target permanent phases out. (Treat it and anything attached to it as though they don't exist until its controllers's next turn.)
```
Comment (kevlahnota) [inline forge-gui/res/adventure/Shandalar/custom_cards/teferis_staff.txt:8]: {M}{M}, {W}{U}

File: forge-gui/res/adventure/Shandalar/custom_cards/nahiris_armory.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Name:Nahiri's Armory
+ManaCost:no cost
+Types:Artifact
+S:Mode$ ReduceCost | ValidCard$ Card | ValidSpell$ Activated.Equip | Activator$ You | Amount$ 1 | EffectZone$ Command | Description$ Equip costs you pay cost {1} less.
+T:Mode$ AttackersDeclared | ValidAttackers$ Creature.modified+YouCtrl | TriggerZones$ Command | Execute$ TrigConjure | TriggerDescription$ Whenever a modified creature you control attacks, conjure a card from Nahiri's Armory into your hand
```
Comment (jjayers99) [inline forge-gui/res/adventure/Shandalar/custom_cards/nahiris_armory.txt:5]: I'm unaware of any cards that do not currently specify these effects as coming from a spellbook. So as awkward as it is, I think we should probably reword this to "Nahiri's Armory's spellbook".

File: forge-gui/res/adventure/Shandalar/custom_cards/nahiris_boss_effect.txt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+Name:Nahiri Boss Effect
+ManaCost:no cost
+Types:Enchantment
+S:Mode$ ReduceCost | ValidCard$ Card | ValidSpell$ Activated.Equip | Activator$ You | Amount$ 1 | EffectZone$ Command | Description$ Equip costs you pay cost {1} less.
+T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Command | Secondary$ True | Execute $ TrigConjure | TriggerDescription$ At the beginning of your upkeep or when an equipped creature you control attacks, conjure a card from Nahiri's Armory into your hand
```
Comment (jjayers99) [inline forge-gui/res/adventure/Shandalar/custom_cards/nahiris_boss_effect.txt:5]: Similar to the comment on Nahiri's Armory, this one is a little weirder still - Either the card should be named Nahiri's Armory and have "'s spellbook" added into this line, or reword it to "Nahiri Boss Effect's spellbook"

File: forge-gui/res/adventure/Shandalar/custom_cards/slimefoots_slimy_staff.txt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+Name:Slimefoot's Slimy Staff
+ManaCost:no cost
+Colors:black,green
+Types:Artifact Equipment
+A:AB$ MakeCard | Cost$ G PayShards<2> | Name$ Bayou | ActivationZone$ Command | Zone$ Hand | IsPresent$ Land.YouOwn | PresentZone$ Hand | PresentCompare$ EQ0 | GameActivationLimit$ 1 | SpellDescription$ Conjure a card named Bayou into your hand. Activate only if there are no land cards in your hand and only once each game. 
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Command | Execute$ TrigPutCounter | TriggerDescription$ At the beginning of your end step, put a spore counter on each fungus you control.
```
Comment (jjayers99) [inline forge-gui/res/adventure/Shandalar/custom_cards/slimefoots_slimy_staff.txt:6]: I've got intentions of reviewing all items with in-match effects to ensure that everything that impacts a match should have a Mana Shard cost to activate - we can tweak the economy suitably later to ensure the players are receiving enough shards to pull the trigger on using these while not being able to hold it down. So while non-binding, I would prefer this to be an activated ability with a shard cost to create this perpetual effect, and to be mutually exclusive with the Bayou conjuring. Keep it as is for now if you like though and we can hammer that out later.

File: forge-gui/res/adventure/Shandalar/custom_cards/teferis_staff.txt [hunk 7]
```diff
@@ -0,0 +1,7 @@
+Name:Teferi's Staff
+ManaCost:no cost
+Types:Artifact Equipment
+A:AB$ Phases | Cost$ U W PayShards<2> | ValidTgts$ Permanent | ActivationZone$ Command | SpellDescription$ Target permanent phases out. (Treat it and anything attached to it as though they don't exist until its controllers's next turn.)
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | TriggerZones$ Command | Execute$ TrigUntap | TriggerDescription$ At the beginning of your end step, untap target artifact or land you control.
```
Comment (jjayers99) [inline forge-gui/res/adventure/Shandalar/custom_cards/teferis_staff.txt:5]: See comments on Slimefoot's Slimy Staff.

**Comment by jjayers99** [comment]
Passing along comments from PR 2493 (https://github.com/Card-Forge/forge/pull/2493#discussion_r1110798843), Exile should be moved from activation cost to ability text on any applicable cards. 

To prevent confusion and clean up cards, I would also recommend that we remove all subtypes from command zone items, and remove any references to the Adventure Mode item that provided the card - I'm currently doing the same to anything I've already added.

**Review by jjayers99 — COMMENTED** [review]
I know you're not done with Garruk, but would recommend having having the life total reset mentioned in the Phase 1 card.

Regarding the sprites / shaky appearance, pretty much everything we're currently using is divided up into 16x16 frames, which makes it really easy to do the math for atlas definitions and to draw sprites that appear relatively steady. The 16x16 format is rather limiting, but that does result in a relatively consistent appearance. It's not a hard constraint that I'm aware of, but I haven't tried anything that doesn't have that layout. Do these sprites appear as expected on the map and match consistently enough with existing art, especially in the "Versus" popup before a match?

**Comment by Simisays** [comment]
> To prevent confusion and clean up cards, I would also recommend that we remove all subtypes from command zone items, and remove any references to the Adventure Mode item that provided the card - I'm currently doing the same to anything I've already added.

I agree with the references to Adventure mode. I've personally used Enchantment for boss effects and equipment for items. I don't think this Forge supports card-type less cards ?

**Comment by kevlahnota** [comment]
looks good, is this ready?

**Review by kevlahnota — APPROVED** [review]
